app/src/main/java/com/example/photofy/PostsAdapter.java

```diff
@@ -0,0 +1,223 @@
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageButton;
+import android.widget.ImageView;
+import android.widget.SeekBar;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.bumptech.glide.Glide;
+import com.example.photofy.models.Photo;
+import com.example.photofy.models.Post;
+import com.example.photofy.models.Song;
+import com.parse.GetCallback;
+import com.parse.Parse;
+import com.parse.ParseException;
+import com.parse.ParseFile;
+import com.parse.ParseObject;
+import com.parse.ParseUser;
+
+import java.io.IOException;
+import java.sql.Time;
+import java.text.DecimalFormat;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+public class PostsAdapter extends RecyclerView.Adapter<PostsAdapter.ViewHolder>{
+
+    public static final String TAG = "PostsAdapter";
+    private Context context;
+    private List<Post> posts;
+
+    public PostsAdapter(Context context, List<Post> posts) {
+        this.context = context;
+        this.posts = posts;
+    }
+
+    @NonNull
+    @Override
+    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(context).inflate(R.layout.item_post, parent, false);
+        return new ViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
+        Post post = posts.get(position);
+        holder.bind(post);
+    }
+
+    @Override
+    public int getItemCount() {
+        return posts.size();
+    }
+
+    // Clean all elements of the recycler
+    public void clear() {
+        posts.clear();
+        notifyDataSetChanged();
+    }
+
+    class ViewHolder extends RecyclerView.ViewHolder {
+
+        private ImageView ivImage;
+        private TextView tvUsername;
+        private TextView tvCaption;
+
+        private TextView tvSongName;
+        private SeekBar seekBar;
+        private TextView tvTime;
+        private ImageButton ibPlay;
+
+        private String duration;
+        private MediaPlayer mediaPlayer;
+        private ScheduledExecutorService timer;
+
+        public ViewHolder(@NonNull View itemView) {
+            super(itemView);
+
+            ivImage = itemView.findViewById(R.id.ivImage);
+            tvUsername = itemView.findViewById(R.id.tvUsername);
+            tvCaption = itemView.findViewById(R.id.tvCaption);
+
+            tvSongName = itemView.findViewById(R.id.tvSongName);
+            seekBar = itemView.findViewById(R.id.seekBar);
+            tvTime = itemView.findViewById(R.id.tvTime);
+            ibPlay = itemView.findViewById(R.id.ibPlay);
+        }
+
+        public void bind(Post post) {
+            Photo picture = (Photo) post.getPhoto();
```

Comment:
this use of slightly different name for variable is a bit confusing, IMO `Photo photo` is a better name for the variable than `picture`